src/snowflake/cli/plugins/object/manager.py

```diff
@@ -52,3 +55,85 @@ def object_exists(self, *, object_type: str, name: str):
             return True
         except ProgrammingError:
             return False
+
+    def _send_rest_request(
+        self, url: str, method: str, data: Optional[Dict[str, Any]] = None
+    ):
+        # SnowflakeRestful.request assumes that API response is always a dict,
+        # which is not true in case of this API, so we need to do this workaround:
+        from snowflake.connector.network import (
+            CONTENT_TYPE_APPLICATION_JSON,
+            HTTP_HEADER_ACCEPT,
+            HTTP_HEADER_CONTENT_TYPE,
+            HTTP_HEADER_USER_AGENT,
+            PYTHON_CONNECTOR_USER_AGENT,
+        )
+
+        self._log.debug(f"Sending {method} request to {url}")
+        rest = self._conn.rest
+        full_url = f"{rest.server_url}{url}"
+        headers = {
+            HTTP_HEADER_CONTENT_TYPE: CONTENT_TYPE_APPLICATION_JSON,
+            HTTP_HEADER_ACCEPT: CONTENT_TYPE_APPLICATION_JSON,
+            HTTP_HEADER_USER_AGENT: PYTHON_CONNECTOR_USER_AGENT,
+        }
+        return rest.fetch(
+            method=method,
+            full_url=full_url,
+            headers=headers,
+            token=rest.token,
+            data=json.dumps(data if data else {}),
+            no_retry=True,
+        )
+
+    def _url_exists(self, url):
+        try:
+            result = self._send_rest_request(url, method="get")
```

Comment:
Instead of `get` you could use `head`.

Reply:
I tried - our REST API seem to not support HEAD requests (it always return 404)